rust/windows-client/src-tauri/src/client/crash_handling.rs

```diff
@@ -0,0 +1,172 @@
+//! A module for handling crashes and writing minidump files
+//!
+//! Mostly copied from <https://github.com/EmbarkStudios/crash-handling/blob/main/minidumper/examples/diskwrite.rs>
+//!
+//! TODO: Capture crash dumps on panic.
+
+use anyhow::{anyhow, bail, Context};
+use known_folders::{get_known_folder_path, KnownFolder};
+use std::{fs::File, io::Write, path::PathBuf};
+
+const SOCKET_NAME: &str = "dev.firezone.client.crash_handler";
+
+use minidumper::{Client, Server};
+
+/// Attaches a crash handler to the client process
+///
+/// Returns a CrashHandler that must be kept alive until the program exits.
+/// Dropping the handler will detach it.
+#[cfg(debug_assertions)]
+pub(crate) fn attach_handler() -> anyhow::Result<crash_handler::CrashHandler> {
+    attach_handler_inner()
+}
+
+#[cfg(not(debug_assertions))]
+pub(crate) fn attach_handler() -> anyhow::Result<crash_handler::CrashHandler> {
+    // Prevent cargo from complaining
+    if false {
```

Comment:
It would say that `attach_handler_inner` is dead code since the release builds wouldn't call it

Reply:
rather than this then we should `#[allow(dead_code)]` for for `attach_handler_inner` or maybe just don't have an inner one?

Reply:
I pushed a commit just now to try to remove the inner function.

I wanted to keep the `target_os` marker there even though the entire client is stubbed out on non-Windows platforms, just so we know the code will compile on other platforms but not work as intended.

And since it's stubbed out, it actually should be fine if this specific module doesn't even compile on other platforms.